index.js

```diff
@@ -123,6 +123,40 @@ function fastifyMultipart (fastify, options, done) {
         req.body = body
       }
     })
+
+    // The following is not available on old Node.js versions
+    // so we must skip it in the test coverage
+    /* istanbul ignore next */
+    if (globalThis.FormData) {
+      fastify.decorateRequest('formdata', async function () {
+        const formData = new FormData()
+        for (const key in this.body) {
+          const value = this.body[key]
+          if (Array.isArray(value)) {
+            for (const item of value) {
+              await append(key, item)
+            }
+          } else {
+            await append(key, value)
+          }
+        }
+
+        async function append (key, entry) {
+          if (entry.type === 'file') {
```

Comment:
This won't work with `attachFieldsToBody: 'keyValues'`

```suggestion
          if (entry.type === 'file' || (attachFieldsToBody === 'keyValues' && Buffer.isBuffer(entry))) {
```

`'keyValues'` comparison could be dropped for performance, it should be fine, or you could just check if it's type is string here

But it becomes harder to understand